test/integration/editor/index.spec.ts

```diff
@@ -43,7 +43,7 @@ test('can move elements in page', async ({ page, browserName }) => {
   const app = await homeModel.createApplication({ dom: domInput });
   await editorModel.goto(app.id);
 
-  await editorModel.pageRoot.waitFor();
+  await editorModel.waitForOverlay();
```

Comment:
@apedroferreira I finally figured out why specs failed on my machine. It seems that my machine might be executing steps "too fast" or at least that's the main assumption. I also find some relevant discussions on playwright repo https://github.com/microsoft/playwright/issues/14752

For some reason CI and probably for the rest of us specs executed at the "right speed" and this issue wasn't visible. On my machine the overlay might not have been "ready" to be clicked on which resulted on "misclicks" and specs not working.

Reply:
I see, so I guess we need to be able to wait for the overlay to be interactive?
Waiting for the overlay to show and having logic related to the overlay at all is an implementation detail that ideally we should avoid - I wonder if we could wait for something more visible to show?

Maybe we can wait for the element we want to interact with to be visible instead (e.g. the button or text field that the test clicks/types in first)? Not sure if it would work though...

Reply:
@apedroferreira I've tried waiting for an actual element, but it seems that element becomes interactive before the overlay appears. If I try to "click center" of the element it simply focuses the actual dom element (i.e. text input) without making a dom selection. If you have some ideas how we could do it otherwise without explicitly waiting for overlay, I'm glad to implement it, but so far I couldn't find a different way

Reply:
I see, can't really think of a better way... So I guess this is the best we can do?

Reply:
I suggest we keep an eye on this and see if we can find a better way 🤔